test/avocado/machineslib.py

```diff
@@ -0,0 +1,166 @@
+
+    def create_vm(self, name, graphics='spice', ptyconsole=False, state='running', wait=False):
+        self.virshvm = name
+
+        img = "/var/lib/libvirt/images/cirros.qcow2"
+        pool_name = "images"
+        pool_path = os.path.dirname(img)
+        self.machine.execute("test -f {}".format(img))
+        self.machine.execute(
+            "sudo virsh pool-list --all | grep {pool} || sudo virsh pool-create-as {pool} --type dir --target {path} && sudo virsh pool-refresh {pool}".format(pool=pool_name, path=pool_path))
+
+        args = {
+            "name": name,
+            "poolName": pool_name,
+            "poolPath": pool_path,
+            "image": img,
+            "logfile": None,
+            "console": "",
+            "graphics": ""
+        }
+
+        if ptyconsole:
+            args["console"] = PTYCONSOLE_XML
+        else:
+            self.machine.execute("sudo chmod 777 /var/log/libvirt")
+            args["logfile"] = "/var/log/libvirt/console-{0}.log".format(name)
+            args["console"] = CONSOLE_XML.format(log=args["logfile"])
+
+        if graphics == 'spice':
+            cxml = SPICE_XML
+        elif graphics == 'vnc':
+            cxml = VNC_XML
+        else:
+            cxml = ""
+        args["graphics"] = cxml
+
+        xml = DOMAIN_XML.format(**args)
+        self.machine.execute('sudo echo \"{}\" > /tmp/xml && sudo virsh define /tmp/xml'.format(xml))
+        if state == 'running':
+            self.machine.execute('sudo virsh start {}'.format(name))
+            if wait:
+                self.wait_vm_complete_start(args)
+        elif state == 'shut off':
+            pass
+
+        self.wait_in_text(name, '#vm-{}-row'.format(name))
+        self.wait_in_text(state, '#vm-{}-state'.format(name))
+        self.click(self.wait_css("tbody tr[data-row-id='vm-{}'] th".format(name), cond=clickable))
+
+        return args
+
+    def wait_vm_complete_start(self, vmargs):
+        log_file = vmargs.get('logfile')
+        if log_file != None:
+            search_str = "login as 'cirros' user."
+            cmd = 'until sudo cat {} | grep "{}"; do sleep 1; done'.format(log_file, search_str)
+            # self.machine.execute(script=WAIT_VM.format(cmd))
+            self.machine.execute(cmd)
+        else:
+            time.sleep(10)
+
+    def destroy_vm(self, name):
+        try:
+            ret = self.machine.execute('sudo virsh domstate {}'.format(name))
+            vmstate = ret.split('\n')[0]
+            if vmstate == 'running':
+                self.machine.execute('sudo virsh destroy {}'.format(name))
+            self.machine.execute('sudo virsh undefine {}'.format(name))
+        except Exception:
+            pass
+
+    def setUp(self):
+        super().setUp()
+        self.login()
+        self.click(self.wait_link('Virtual Machines', cond=clickable))
+        self.wait_frame("machines")
+
+    def tearDown(self):
+        super().tearDown()
+        if hasattr(self, 'virshvm'):
```

Comment:
maybe ``virshvm`` could be set in ``setUp`` method to ``None`` and then just do simpler ``if`` there.

Reply:
Makes sense.